src/ast/document-descriptor.ts

```diff
@@ -38,3 +46,206 @@ export class DocumentDescriptor implements Descriptor {
     return this.document.url;
   }
 }
+
+export interface Feature {
+  kinds: Iterable<string>;
+  identifiers?: Iterable<string>;
+  // sourceLocation: SourceLocation;
+}
+
+
+export interface Resolvable extends ScannedFeature {
+  resolve(document: Document): Feature;
+}
+function isResolvable(x: any): x is Resolvable {
+  return x.resolve && typeof x.resolve === 'function';
+}
+
+export class Document implements Feature {
+  url: string;
+  parsedDocument: ParsedDocument<any, any>;
+  isInline: boolean;
+
+  kinds: Iterable<string>;
+  identifiers: Set<string>;
+
+  private _rootDocument: Document;
```

Comment:
what happens when we have multiple root documents? A single dependency can be referenced from multiple entry points. Imagine we add a version of `analyze()` that takes an array for convenience, and we pass the entry points: `analyzer.analyze(['a.html', 'b.html'])` and `c.html` is referenced somewhere from `a` and `b`. What's the root of `c.html`?